### PR TITLE
ci: setup a config per package

### PR DESCRIPTION
**Pull request by BioPhoton**
This PR adds a `code-pushup.config.ts` to a couple of packages.

**Depends on:**
#735 
#737 

**Included changes:**
- [ ] ✏️ edit `code-pushup.config.ts` at root with only plugins `lighthouse`, `js-packages`
- [ ] ➕ add `code-pushup.config.ts` to 📦 `models` with plugins `eslint`, `js-packages`
- [ ] ➕ add `code-pushup.config.ts` to 📦 `utils` with plugins `eslint`, `js-packages`
- [ ] ➕ add `code-pushup.config.ts` to 📦 `core` with plugins `eslint`, `js-packages`
- [ ] ➕ add `code-pushup.config.ts` to 📦 `cli` with plugins `eslint`, `js-packages`
- [ ] ➕ add `code-pushup.config.ts` to 📦 `plugin-eslint` with plgins `eslint`, `js-packages`
- [ ] ➕ add `code-pushup.config.ts` to 📦 `plugin-lighthouse` with plugins `eslint`, `js-packages` 
- [ ] ➕ add `code-pushup.config.ts` to 📦 `plugin-coverage` with plgins `eslint`, `js-packages`

### CODE REVIEW

**Comment by vmasek** [comment]
We could merge it like this and I'll simplify it later with `mergeConfig` helper

**Review by vmasek — APPROVED** [review]
LGTM, I'd wait for @matejchalk for final approval tho 🤔

File: packages/cli/code-pushup.config.ts
Comment (matejchalk) [inline packages/cli/code-pushup.config.ts:4]: You realize this doesn't give any type-safety, right? 

image

The package obviously isn't (and shouldn't be) installed, and for the TS path aliases to work the `code-pushup.config.ts` would have to part of some `tsconfig` (would require a special `tsconfig.code-pushup.json`, which is probably overkill). Simple solution is to import type from source directly:

```suggestion
import type { CoreConfig } from '../models/src';
```

File: packages/cli/code-pushup.config.ts
Comment (matejchalk) [inline packages/cli/code-pushup.config.ts:38]: Invalid property name:

image


```suggestion
      packageJsonPaths: [join('packages', 'cli', 'package.json')],
```

---

However, I don't think it makes sense to run JS packages plugin per project. It still runs `npm audit`, etc. on the root `package-lock.json`, so the results are the same for each project. E.g. outdated `lighthouse` is reported for `cli` project, even though `cli` doesn't use it.

**Review by matejchalk — CHANGES_REQUESTED** [review]

File: e2e/cli-e2e/code-pushup.config.ts
Comment (BioPhoton) [reply e2e/cli-e2e/code-pushup.config.ts:32]: It will end up in the executor #737

File: packages/cli/code-pushup.config.ts
Comment (BioPhoton) [reply packages/cli/code-pushup.config.ts:38]: Wouldnt it help if we know per project what we release?

**Pull request closed without merging**
